polkadot/cli/src/lib.rs

```diff
@@ -85,9 +79,9 @@ pub fn run<I, T, W>(args: I, worker: W, version: cli::VersionInfo) -> error::Res
 	W: Worker,
 {
 
-	match cli::prepare_execution::<service::Factory, _, _, _, _>(args, worker.exit_only(), version, load_spec, "parity-polkadot")? {
+	match cli::prepare_execution::<service::Factory, _, _, _, _>(args, worker, version, load_spec, "parity-polkadot")? {
 		cli::Action::ExecutedInternally => (),
-		cli::Action::RunService(mut config) => {
+		cli::Action::RunService((mut config, worker)) => {
```

Comment:
Hm, I'm not sure how it works. The second item in tuple of `Action::RunService` is bound to be `IntoExit`, why it's called `worker` here? Why `worker.configuration()` works here?

Reply:
`Action` returns back the same thing that was passed to `prepare_execution` if it was not used.